thorin/fe/lexer.cpp

```diff
@@ -39,16 +39,15 @@ Tok Lexer::lex() {
             return *cache;
         }
 
-        loc_.begin = ahead().pos;
-        str_.clear();
+        begin();
```

Comment:
`begin` might not be the best name, as it's quite heavily associated with iterators in C++. maybe prefer `start`?